Modules/Core/Common/test/itkObjectFactoryOnlyNewTest.cxx

```diff
@@ -0,0 +1,206 @@
+  }
+  return true;
+}
+
+
+int
+itkObjectFactoryOnlyNewTest(int, char *[])
+{
+  // Verify that interface class cannot be instantiated
+  // if no implementation factory is registered
+  using PixelType = short;
+  const size_t Dimension = 2;
+  using TestImageInterfaceType = TestImageInterfaceClass<PixelType, Dimension>;
+  ITK_TRY_EXPECT_EXCEPTION(TestImageInterfaceType::New());
+
+  // Register implementation override
+  auto factory = TestFactory::New();
+  itk::ObjectFactoryBase::RegisterFactory(factory);
+
+  // Log all registered factories
+  std::list<itk::ObjectFactoryBase *> factories = itk::ObjectFactoryBase::GetRegisteredFactories();
+  std::cout << "----- Registered factories -----" << std::endl;
+  for (auto & oneFactory : factories)
+  {
+    std::cout << "  Factory version: " << oneFactory->GetITKSourceVersion() << std::endl
+              << "  Factory description: " << oneFactory->GetDescription() << std::endl;
+
+    std::list<std::string>                 overrides = oneFactory->GetClassOverrideNames();
+    std::list<std::string>                 names = oneFactory->GetClassOverrideWithNames();
+    std::list<std::string>                 descriptions = oneFactory->GetClassOverrideDescriptions();
+    std::list<bool>                        enableflags = oneFactory->GetEnableFlags();
+    std::list<std::string>::const_iterator n = names.begin();
+    std::list<std::string>::const_iterator d = descriptions.begin();
+    std::list<bool>::const_iterator        e = enableflags.begin();
+    for (std::list<std::string>::const_iterator o = overrides.begin(); o != overrides.end(); ++o, ++n, ++d, ++e)
+    {
+      std::cout << "    Override " << *o << " with " << *n << std::endl
+                << "      described as \"" << *d << "\"" << std::endl
+                << "      enabled " << *e << std::endl;
+    }
+  }
+  std::cout << "----- -----" << std::endl;
+
+  TestImageInterfaceType::Pointer v;
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(v = TestImageInterfaceType::New());
+
+  factory->Print(std::cout);
+
+  int status = EXIT_SUCCESS;
+  if (!TestNewImage(v, "TestImageImplementationType"))
+  {
+    status = EXIT_FAILURE;
+  }
+
+  // disable interface class creation through the factory
+  factory->Disable(typeid(TestImageInterfaceType).name());
+  ITK_TRY_EXPECT_EXCEPTION(v = TestImageInterfaceType::New());
+
+  // re-enable interface class creation through the factory
+  using TestImageImplementationType = TestImageImplementationClass<PixelType, Dimension>;
+  factory->SetEnableFlag(true, typeid(TestImageInterfaceType).name(), typeid(TestImageImplementationType).name());
+  ITK_TRY_EXPECT_NO_EXCEPTION(v = TestImageInterfaceType::New());
+  if (!TestNewImage(v, "TestImageImplementationType"))
+  {
+    status = EXIT_FAILURE;
+  }
+
+  // remove factory and verify interface class cannot be created
+  itk::ObjectFactoryBase::UnRegisterFactory(factory);
+  ITK_TRY_EXPECT_EXCEPTION(v = TestImageInterfaceType::New());
+
+  return EXIT_SUCCESS;
```

Comment:
Returning `status` will make CI builds pass.

Reply:
Thanks for the catch, I based this test off of [`itkObjectFactoryTest.cxx`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/test/itkObjectFactoryTest.cxx) but overlooked `status`.

Amended to remove `status` and immediately return `EXIT_FAILURE` if a failure case is encountered.